### PR TITLE
Document license file expiration logic

### LINKED ISSUE #19109: Document Teleport license file management
## Applies To
Probably `https://goteleport.com/docs/deploy-a-cluster/teleport-enterprise/license/`

## Details
Update license file management documentation

- What is the end date of a Teleport Enterprise license
- What happens after the end date
- How to update license file
- What users can expect in terms of warning leading up to expiration date

### PR DESCRIPTION
**Pull request by ptgott**
Closes #19109

### CODE REVIEW

File: docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx
Comment (zmb3) [inline docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx:6]: Should we add a warning at the top when `scope==cloud` to say that cloud licenses are managed for you and this guide does not apply to cloud?

File: docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx
Comment (zmb3) [inline docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx:92]:
```suggestion
or removed support for a feature, you must obtain an update license file and
```

File: docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx
Comment (zmb3) [inline docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx:147]: We're starting to standardize the error messages for enterprise features. In most cases, it will look like this:

```text
SAML: this feature requires Teleport Enterprise
```

**Review by zmb3 — APPROVED** [review]

**Review by stevenGravy — APPROVED** [review]

**Comment by ptgott** [comment]
@alexfornuto Did optimization make a difference here? I thought I had already used `trimage` on the new images

**Comment by ptgott** [comment]
Also waiting for one more review from @xinding33

**Comment by alexfornuto** [comment]
> @alexfornuto Did optimization make a difference here? I thought I had already used `trimage` on the new images

Not really. I didn't realize you had already optimized, though it's obvious once I pushed and saw the difference.

Feel free to drop that commit if you want.

File: docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx
Comment (xinding33) [inline docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx:21]: @russjones do we have a name for the customer portal?

File: docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx
Comment (xinding33) [inline docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx:110]: @ptgott can you clarifying what dismissing the message will do?

It should disable the message for the duration of the session.

**Review by xinding33 — APPROVED** [review]
A few nites but otherwise looks good to me. Thanks!

File: docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx
Comment (ptgott) [reply docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx:21]: Looks like "customer portal" is the most common usage, so I'll change it to that:

```
% grep -RohE "\[[A-Za-z ]+\]\(https://dashboard.gravitational.com/web/login\)" | sort | uniq -c | sort -n
   1 [Dashboard](https://dashboard.gravitational.com/web/login)
   1 [customer dashboard](https://dashboard.gravitational.com/web/login)
   3 [Customer Portal](https://dashboard.gravitational.com/web/login)
  10 [customer portal](https://dashboard.gravitational.com/web/login)
```

**Pull request merged**
